### ISSUE TITLE
Undefined is not a function (evaluating inst.render)

### ISSUE THREAD

**Issue by sdb1228** [mauron85/react-native-background-geolocation#13]
* Platform: Both
* Link to your project: https://github.com/sdb1228/RNbackgroundLocationTest

I am getting a runtime error when I try to paste in the example posted on the README.  I am new to react native so I could have screwed something up when setting up my project.  The Project I posted is just a very basic example

If you run either device you will see a failure.  Android is probably more reliable given the whole sqlite thing.

**Comment by sdb1228** [comment]
I am an idiot figured it out.

**Comment by qingchanghan** [comment]
How did you resolve it?

**Comment by tomatopeel** [comment]
@sdb1228 real bad etiquette to say that and not post the resolution... Having the same problem, guess I'm also an idiot and will have to figure it out the long way now :-)

**Comment by sdb1228** [comment]
Sorry guys! You are right totally unprofessional.  I posted the repo I used above https://github.com/sdb1228/RNbackgroundLocationTest And I am trying to recall how it was broken originally.  If you guys have a repo I can look at I can help debug since I was the original one to open this ticket!

**Comment by tomatopeel** [comment]
I think it's just that this library should be used in the componentWillMount() function on any component of your choice in your project, rather than trying to use a standalone component for this (originally I'd just copy-pasted the example code, but it seems to work if I just copy-paste the code from the componentWillMount() method to one of my components, still need to test this further but no errors are produced and the debug line in the console confirms that it is started).

**Comment by mauron85** [comment]
I can't help here. Original example repo is dead and there is no information how to reproduce this. Anyway, regarding `componentWillMount` my comment might be helpful.

https://github.com/mauron85/react-native-background-geolocation/issues/79#issuecomment-305970565

**Comment by sdb1228** [comment]
@tomatopeel I can reopen this if you can give us a repo with a reproduction.  I would love to help with since I didn't post the solution originally. I apologize again though for not telling you what was the solution was.

**Comment by tomatopeel** [comment]
@sdb1228 no it's fine haha, this error is just caused by the absence of the render function in a component. Thanks for getting back in touch anyway! :)

**Comment by jerinho** [comment]
> Sorry guys! You are right totally unprofessional. I posted the repo I used above https://github.com/sdb1228/RNbackgroundLocationTest And I am trying to recall how it was broken originally. If you guys have a repo I can look at I can help debug since I was the original one to open this ticket!
> 
> Edit: Are you trying to define a render func inside the component that is using Background-geolocator

wow. almost a year before you figured out your misbehavior. and now your brain was cleared out already. hopefully we don't have too many guys like you

**Comment by sdb1228** [comment]
Sorry again @jerinho! Totally understand the unprofessionalism that came here.  Would love to help if you have a repo that has the problem reproduced. Otherwise have you tried another solution in this thread?

**Comment by vimalloc** [comment]
> wow. almost a year before you figured out your misbehavior. and now your brain was cleared out already. hopefully we don’t have too many guys like you. 

...have you ever gone back and looked at code written a long time ago with no current context around it? We are humans here, not machines that remembers everything we did over a year ago. 

> hopefully we don’t have too many guys like you

That attitude is not going to get you anywhere. This is an open source project, where people have written a bunch of code and are letting you use it for free. No one owes you anything here. 

> i'm facing this issue and what's more annoying is google misled me here. I think the best what you should do is provide us the solution or delete this thread

That’s simply not how github and open source works. If you are having an issue with the library, open a new issue. That is way more likely to get seen then necro numbing an issue that has been closed for over 2 years.
